### PR TITLE
ci: generate should include docs-api

### PR DESCRIPTION
**Pull request by zirain**
auto update references

### CODE REVIEW

File: .github/markdown_lint_config.json [hunk 1]
```diff
@@ -48,5 +48,6 @@
   "MD047": false,
   "MD048": false,
   "MD049": false,
-  "MD050": false
+  "MD050": false,
+  "MD051": false
```
Comment (zirain) [inline .github/markdown_lint_config.json:52]: temporarily hack, xref: #1246

File: tools/src/crd-ref-docs/go.mod [hunk 2]
```diff
@@ -2,10 +2,11 @@ module local
 
 go 1.20
 
+replace github.com/elastic/crd-ref-docs => github.com/zirain/crd-ref-docs v0.0.0-20230403084159-e8e051398dc2
```
Comment (zirain) [inline tools/src/crd-ref-docs/go.mod:5]: xref: https://github.com/elastic/crd-ref-docs/pull/37

**Comment by arkodg** [comment]
thanks for raising this @zirain 
looks like the core fix tied to the PR title is
```
generate: kube-generate go.generate docs-api
```

would it be possible to split this up into 2 PRs so the other useful fix of linking upstream Gateway API docs can be brought in later once the PR you raised has made its way into the upstream repo ?

**Comment by arkodg** [comment]
you can circumvent the situation by ignoring the field for now
https://github.com/envoyproxy/gateway/blob/793595957fb9348fd64960aa018e1e20a390bcbf/tools/crd-ref-docs/config.yaml#L4

File: .github/markdown_lint_config.json (hunk 1 above)
Comment (zirain) [inline .github/markdown_lint_config.json:52]: add this line to circumvent lint error

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**